### PR TITLE
test/bench_log: Add a function: void usage(char *);

### PR DESCRIPTION
**Pull request by chenxuqiang**
Currently, the bench_log does not have a usage function, so that when it is used, if no pararmeters are given, a segmentation error will occur. This causes confusion.

Now, we add the usage function and add (argc < 3) conditional judgment.

Signed-off-by: Xuqiang Chen <chenxuqiang3@hisilicon.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/test/bench_log.cc [hunk 1]
```diff
@@ -30,8 +30,19 @@ struct T : public Thread {
...
+  cout << program_name << " <threads> <lines>" << std::endl;
+  cout << "\t threads: the number of thtreads for this test." << std::endl;
+  cout << "\t lines: the number of the lines per thread." << std::endl;
+}
+
 int main(int argc, const char **argv)
 {
+  if (argc < 3) {
+    usage(argv[0]);
+    return 0;
```
Comment (tchaikov) [inline src/test/bench_log.cc]: nit, could return `EXIT_FAILURE` here.

File: src/test/bench_log.cc [hunk 2]
```diff
@@ -30,8 +30,19 @@ struct T : public Thread {
   }
 };
 
+void usage(const char* program_name) {
+  cout << program_name << " <threads> <lines>" << std::endl;
+  cout << "\t threads: the number of thtreads for this test." << std::endl;
```
Comment (tchaikov) [inline src/test/bench_log.cc]: typo.

File: src/test/bench_log.cc [hunk 3]
```diff
@@ -30,8 +30,19 @@ struct T : public Thread {
...
+  cout << "\t lines: the number of the lines per thread." << std::endl;
```
Comment (tchaikov) [inline src/test/bench_log.cc]: s/lines/log entries/

File: src/test/bench_log.cc (hunk 1 above)
Comment (chenxuqiang) [reply src/test/bench_log.cc]: OK

File: src/test/bench_log.cc [hunk 4]
```diff
@@ -30,8 +30,19 @@ struct T : public Thread {
   }
 };
 
+void usage(const char* program_name) {
+  cout << program_name << " <threads> <lines>" << std::endl;
+  cout << "\t threads: the number of threads for this test." << std::endl;
+  cout << "\t lines: the number of log entries per thread." << std::endl;
```
Comment (tchaikov) [inline src/test/bench_log.cc]: nit, i'd suggest put something like
```c++
  cout << program_name << " <threads> <lines>"
       << "\t threads: the number of threads for this test."
       << "\t lines: the number of log entries per thread."
       << std::endl;
```

more compacted and concise this way.

**Comment by tchaikov** [comment]
@chenxuqiang could you revise the title 

```
test/bench_log: Add a function: void usage(char *);
```
like
```
test/bench_log: print usage if number of argument is less than 2
```

also could you squash these two commits into a single one?

File: src/test/bench_log.cc (hunk 4 above)
Comment (chenxuqiang) [reply src/test/bench_log.cc]: OK

File: src/test/bench_log.cc [hunk 5]
```diff
@@ -30,8 +30,20 @@ struct T : public Thread {
   }
 };
 
+void usage(const char* program_name) {
+  cout << program_name << " <threads> <lines>\n"
+       << "\t threads: the number of threads for this test\n."
+  	   << "\t lines: the number of log entries per thread.\n"
```
Comment (tchaikov) [inline src/test/bench_log.cc]: might want use spaces to be consistent with other lines in this very function?

**Comment by tchaikov** [comment]
also could you squash these four commits into a single one?

**Comment by tchaikov** [comment]
@chenxuqiang why did you close this PR?

**Comment by tchaikov** [comment]
see also #31723

**Pull request closed without merging**